samza-rest/src/main/java/org/apache/samza/rest/script/ScriptRunner.java

```diff
@@ -84,13 +90,21 @@ public int runScript(String scriptPath, ScriptOutputHandler outputHandler, Strin
    * @param args        the command line args to pass to the script.
    * @return            a {@link java.lang.ProcessBuilder} for the script and args.
    */
-  private ProcessBuilder getProcessBuilder(String scriptPath, String[] args) {
+  private ProcessBuilder getProcessBuilder(String scriptPath, String[] args) throws FileNotFoundException {
+    if (!new File(scriptPath).exists()) {
+      throw new FileNotFoundException("Script file does not exist: " + scriptPath);
+    }
+
     List<String> command = new ArrayList<>(args.length + 1);
     command.add(scriptPath);
     command.addAll(Arrays.asList(args));
 
     log.debug("Building process with command {}", command);
-    return new ProcessBuilder(command);
+    ProcessBuilder pb =  new ProcessBuilder(command);
+
+    pb.environment().clear();
+    pb.environment().putAll(environment);
```

Comment:
Minor suggestion(Please change if it makes sense): 

Just wondering, if it's okay to take in environmentMap as runScript param(just like commandArgs or even as constructor param) and use it for processcreation(if we want to support client of this api running script with custom env).

I think how setting environment  will affect process creation will be simple that way. Otherwise  a client wanting to modify env will have do getMap() and modify it and then call runScript.  

Or let's disallow clients to even modify envMap(hiding the get method).